### PR TITLE
fix(server): remove catalogs

### PR DESCRIPTION
**Pull request by jlahovnik**
removes the /catalogs endpoints and the corresponding logic, config and documentation from the server mode

### CODE REVIEW

**Review by dalpasso — APPROVED** [review]

**Review by sbrunato — DISMISSED** [review]
Please
- [x] do not list `/catalogs/*` or `/collections/*` in landing page
- [x] remove `/catalogs` feature from `eodag/resources/stac.yml` and `eodag/rest/stac.py`
- [x] update documentation

**Comment by jlahovnik** [comment]
> Please
> 
>     * [x]  do not list `/catalogs/*` or `/collections/*` in landing page
> 
>     * [x]  remove `/catalogs` feature from `eodag/resources/stac.yml` and `eodag/rest/stac.py`
> 
>     * [x]  update documentation

catalogs endpoints and correspoding logic removed

**Review by dalpasso — CHANGES_REQUESTED** [review]
1. Update/remove reference to `catalogs`:

https://github.com/CS-SI/eodag/blob/fdf0f1b14c367c8dcf4ce456d43114549eae0448/README.rst?plain=1#L177-L188

2. Update URLs to use `collections` instead of `catalogs`:

https://github.com/CS-SI/eodag/blob/fdf0f1b14c367c8dcf4ce456d43114549eae0448/docs/stac_rest.rst?plain=1#L113-L124

3. Probably this could be removed:

https://github.com/CS-SI/eodag/blob/fdf0f1b14c367c8dcf4ce456d43114549eae0448/eodag/rest/constants.py#L27

4. Parameters `catalogs` should be changed to `collection` or `collection_id`:

https://github.com/CS-SI/eodag/blob/fdf0f1b14c367c8dcf4ce456d43114549eae0448/eodag/rest/core.py#L234-L241

5. Update documentation:

https://github.com/CS-SI/eodag/blob/fdf0f1b14c367c8dcf4ce456d43114549eae0448/eodag/rest/stac.py#L871

**Comment by jlahovnik** [comment]
catalogs removed from download and download docs

**Review by dalpasso — APPROVED** [review]

**Review by sbrunato — DISMISSED** [review]

File: docs/stac_rest.rst [hunk 1]
```diff
@@ -75,43 +75,6 @@ Example URL:
...
-
-(``EODAG_LOGGING`` environment variable definition will increase ``eodag``
-logging level, and accepts values: 1, 2, or 3 for maximum level)
-
-And browse http://127.0.0.1:5001:
-
-.. image:: _static/stac_browser_example.png
-   :width: 800
-   :alt: STAC browser example
-
```
[the remark this answers is not in the document]
Comment (jlahovnik) [reply docs/stac_rest.rst]: STAC Browser section re-added

**Review by sbrunato — CHANGES_REQUESTED** [review]
browsing by catalog still appears in`README`, image https://raw.githubusercontent.com/CS-SI/eodag/develop/docs/_static/stac_browser_example_600.png.
Please:
- remove this image (resized and full)
- replace this outdated image and `stac_browser_start.png` with an nicer one (full and resized), similar to `stac_browser_example.png` (using a search)

File: docs/_static/stac_browser_start.png
Comment (sbrunato) [inline docs/_static/stac_browser_start.png:1]: should be deleted

**Review by sbrunato — CHANGES_REQUESTED** [review]

**Review by sbrunato — APPROVED** [review]

**Pull request merged**